src/main/java/com/zendesk/maxwell/bootstrap/MaxwellBootstrapUtility.java

```diff
@@ -151,22 +151,37 @@ private Long getTotalRowCount(Connection connection, Long bootstrapRowID) throws
 		}
 	}
 
-	private Long calculateRowCount(Connection connection, String db, String table) throws SQLException {
+	private Long calculateRowCount(Connection connection, String db, String table, String whereClause) throws SQLException {
 		LOGGER.info("counting rows");
 		String sql = String.format("select count(*) from `%s`.%s", db, table);
+		if ( whereClause != null ) {
+			sql += String.format(" where %s", whereClause);
+		}
 		PreparedStatement preparedStatement = connection.prepareStatement(sql);
 		ResultSet resultSet = preparedStatement.executeQuery();
 		resultSet.next();
 		return resultSet.getLong(1);
 	}
 
-	private long insertBootstrapStartRow(Connection connection, String db, String table, Long totalRows) throws SQLException {
+	private long insertBootstrapStartRow(Connection connection, String db, String table, String whereClause, Long totalRows) throws SQLException {
 		LOGGER.info("inserting bootstrap start row");
-		String sql = "insert into `bootstrap` (database_name, table_name, total_rows) values(?, ?, ?)";
+		String sql = null;
+		if ( whereClause != null ) {
```

Comment:
I think you can make the SQL unconditionally include the column, just go ahead and pass `null` to the `where` column on line 179 no?